lib/OpenIDConnect/Repositories/ClientRepository.php

```diff
@@ -22,7 +22,9 @@ public function getClientEntity($clientIdentifier)
         $client = new ClientEntity();
         $client->setIdentifier($tool->oidc_client_id);
         $client->setName($tool->name);
-        $client->setRedirectUri($tool->oidc_redirect_url);
+        $client->setRedirectUri(
```

Comment:
I would suggest **not** to do the `explode` directly in the `setRedirectUri`, but to do it beforehand with empty array handler in order to make sure that the passing argument to the method is always an array!


```suggestion
        $redirect_urls_arr = $tool->oidc_redirect_url ? explode("\n", $tool->oidc_redirect_url) : [];
        $client->setRedirectUri($redirect_urls_arr);
```

Reply:
I don't like the idea to pass an empty array as default because the `oidc_redirect_url` parameter is required in OIDC and imo an empty array makes no sense. I would suggest to simply pass the value of `$tool->oidc_redirect_url` instead because the `setRedirectUri` function allows string and array values.

```suggestion
        $redirect_urls = $tool->oidc_redirect_url ? explode("\n", $tool->oidc_redirect_url) : $tool->oidc_redirect_url;
        $client->setRedirectUri($redirect_urls);
```

Alternatively, we could throw an error if no value is present.

Reply:
It seems that the `oauth2-server` package implementation contains an error handling for empty array and strings in AuthCodeGrant.php.